content/2017.12.kinto-at-mozilla.rst

```diff
@@ -0,0 +1,148 @@
+Kinto at Mozilla
+################
+
+:lang: en
+:date: 2017-12-20
+:url: kinto-at-mozilla
+:summary: A summary of how Kinto is used at Mozilla
+
+
+During the last company «all hands», we've heard many mentions of *Kinto*, mostly from people that we hadn't met before or that wasn't aware we were the authors. Part of this fame may probably be due to the success of Firefox Notes, and the recent addition of the long awaited synchronization features.
+
+It's amazing how we've accomplished with this project in very heterogenous contexts, and sadly we haven't really taken the time to celebrate. This post intends to present the different use-cases we are aware of at Mozilla, with some architecture and metrics details.
+
+
+Firefox remote settings
+=======================
+
+The historical one!
+
+Firefox engineers edit remote settings like blocklist or certificates revocation lists via the Kinto Admin UI. There are several buckets for each category of settings listed below.
+
+The hundreds millions of Firefox browsers regularly poll for changes using the `kinto-changes` endpoint. Two instances are deployed with the same database, a public one read-only behind a CDN, and a private one only accessible via a secured VPN.
+
+Since the remote settings are critical, and served via a CDN, we use `kinto-signer` to sign the data in order to guarantee authenticity and integrity. It also brings reviewing and sign-off features as well as previewing changes for the QA team. The history plugin allows to track the changes over time.
+
+The client lies in the Firefox codebase, implemented with `kinto.js`. It verifies `content signatures` to verify that settings are conform, and has a specific adapter to store local data in SQlite instead of IndexedDB. Some `Telemetry data` is sent when remote settings are synchronized locally in order to track uptake globally.
+
+In order to manage buckets and collections, as well as groups and their related permissions, we use `kinto-wizard` that reads a YAML file from a Github private repo, and creates or updates the missing objects.
+
+Instance: https://firefox.settings.services.mozilla.com
+Production date: December 2015
+Average requests/sec: 10 (*behind the Cloudfront CDN*)
+
+Blocklists
+----------
+
+The blocklists allow to remotely disable malware addons, malicious or unstable plugins, or some graphical drivers that come to be unstable on certain hardware. A custom plugin, kinto-amo, was implemented to serve the blocklist in the same XML format as addons.mozilla.org used to have — for old Firefox versions.
+
+Certificates revocation
```

Comment:
`Certificate revocation` again.

In my experience, I've seen you use plurals in compound noun phrases where it should be singulars more than the other way around. I thought that was because it should always be singulars but I guess there are [complicated rules](http://www.quickanddirtytips.com/education/grammar/compound-nouns).